### PR TITLE
Fix error messages for brew edit and brew create

### PR DESCRIPTION
**Pull request by bayandin**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

- Fix `brew edit` error message if the formula doesn't exist (`brew create` doesn't have `--formula` argument)
- Fix `brew create` error message  if run without `--tap` argument and the default tap/cask is untapped

### CODE REVIEW

File: Library/Homebrew/dev-cmd/create.rb [hunk 1]
```diff
@@ -94,7 +94,7 @@ def create_cask(args:)
     token = Cask::Utils.token_from(name)
 
     cask_tap = Tap.fetch(args.tap || "homebrew/cask")
-    raise TapUnavailableError, args.tap unless cask_tap.installed?
+    raise TapUnavailableError, cask_tap unless cask_tap.installed?
```
Comment (Bo98) [inline Library/Homebrew/dev-cmd/create.rb:97]: Not that it matters in practice, but this is more consistent with how this error is used elsewhere:

```suggestion
    raise TapUnavailableError, cask_tap.name unless cask_tap.installed?
```

File: Library/Homebrew/dev-cmd/create.rb [hunk 2]
```diff
@@ -151,7 +151,7 @@ def create_formula(args:)
     fc.version = args.set_version
     fc.license = args.set_license
     fc.tap = Tap.fetch(args.tap || "homebrew/core")
-    raise TapUnavailableError, args.tap unless fc.tap.installed?
+    raise TapUnavailableError, fc.tap unless fc.tap.installed?
```
Comment (Bo98) [inline Library/Homebrew/dev-cmd/create.rb:154]:
```suggestion
    raise TapUnavailableError, fc.tap.name unless fc.tap.installed?
```

**Review by Bo98 — COMMENTED** [review]
Thanks!

**Review by Bo98 — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks @bayandin!

**Pull request merged**